tests/test_accessors.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+import spherely
+
+import pytest
+
+
+@pytest.mark.parametrize(
+    "geog, expected",
+    [
+        (spherely.Point(0, 0), spherely.Point(0, 0)),
+        (
+            spherely.LineString([(0, 0), (0, 2)]),
+            spherely.Point(0, 1),
+        ),
+        (
+            spherely.Polygon([(0, 0), (2, 0), (2, 2), (0, 2)]),
+            spherely.Point(1, 1),
+        ),
+    ],
+)
+def test_centroid(geog, expected) -> None:
+    # scalar
+    actual = spherely.centroid(geog)
+    assert isinstance(actual, spherely.Point)
+    # TODO add some way of testing almost equality
+    # assert spherely.equals(actual, expected)
```

Comment:
Testing is not super easy at the moment (we will have to develop some utilities to help with this). 

Currently this gives failures like

```
>       assert spherely.equals(actual, expected)
E       assert False
E        +  where False = <built-in method equals of PyCapsule object at 0x7fea1738e790>(POINT (4.999999999999999 0), POINT (5 0))
E        +    where <built-in method equals of PyCapsule object at 0x7fea1738e790> = spherely.equals
```

So either we need to have a version of `equals` that does some rounding/snapping of coordinates (I think S2 should provide some options for that, which we probably want to expose). 
Or more generally test based on the WKT repr, but then we also need to have a way to specify / reduce the precision of the WKT repr (https://github.com/benbovy/spherely/issues/8)

Reply:
I agree we need better helpers or features for testing (getting object coordinates would be useful too). This is short-term priority I think.